tools/dockerfile-llb-frontend/parse/parse.go

```diff
@@ -0,0 +1,76 @@
+// Package parse implements the logic for reading and manipulating kraftfiles.
+package parse
+
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2022, Unikraft GmbH and The KraftKit Authors.
+// Licensed under the BSD-3-Clause License (the "License").
+// You may not use this file except in compliance with the License.
+
+import (
+	"fmt"
+	"os"
+
+	"gopkg.in/yaml.v3"
+)
+
+// Kraftfile is a representation of a config loaded from kraft.yaml.
+type Kraftfile struct {
+	Specification string                 `yaml:"specification"`
+	Name          string                 `yaml:"name"`
+	Outdir        string                 `yaml:"outdir"`
+	Template      map[string]interface{} `yaml:"template"`
+	Unikraft      unikraft               `yaml:"unikraft"`
+	Targets       []target               `yaml:"targets"`
+	Libraries     map[string]library     `yaml:"libraries"`
+}
+
+type target struct {
+	Name         string                 `yaml:"name"`
+	Architecture string                 `yaml:"architecture"`
+	Platform     string                 `yaml:"platform"`
+	Initrd       map[string]interface{} `yaml:"initrd"`
+	Command      []string               `yaml:"command"`
+}
+
+type component struct {
+	Source  string                 `yaml:"source"`
+	Version string                 `yaml:"version"`
+	Kconfig map[string]interface{} `yaml:"kconfig"`
+}
+
+type unikraft struct {
+	component
+}
+
+type library struct {
+	component
+}
```

Comment:
These are already defined within `kraftkit.sh/unikraft/*`, we should rely on these and load the `Kraftfile` via:

```go
// Initialize at least the configuration options for a project
project, err := app.NewProjectFromOptions(ctx,
	app.WithProjectWorkdir(workdir),
	app.WithProjectKraftfile(kraftfile),
)
if err != nil {
	return err
}
```

Reply:
Thanks. I've gotten rid of the parse package and the plugin now uses the core Kraftkit logic.